host-go/engine/tests/wasm32_pipeline_with_state_test.go

```diff
@@ -0,0 +1,99 @@
+package tests
+
+import (
+	"testing"
+
+	"github.com/lens-vm/lens/host-go/engine"
+	"github.com/lens-vm/lens/tests/modules"
+	"github.com/sourcenetwork/immutable/enumerable"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// This test asserts how state can be shared between pipe stages if the same module-instance is
+// appended multiple times.
+func TestWasm32PipelineWithSharedState(t *testing.T) {
+	type Value struct {
+		Id   int
+		Name string
+	}
+	runtime := newRuntime()
+
+	module, err := engine.NewModule(runtime, modules.WasmPath5)
+	if err != nil {
+		t.Error(err)
+	}
+
+	instance, err := engine.NewInstance(module)
+	if err != nil {
+		t.Error(err)
+	}
+
+	source := enumerable.New([]Value{
+		{
+			Name: "John",
+		},
+		{
+			Name: "Shahzad",
+		},
+		{
+			Name: "Addo",
+		},
+	})
+
+	pipe := engine.Append[Value, Value](source, instance)
+	pipe = engine.Append[Value, Value](pipe, instance)
+	pipe = engine.Append[Value, Value](pipe, instance)
+
+	hasNext, err := pipe.Next()
+	if err != nil {
+		t.Error(err)
+	}
+	assert.True(t, hasNext)
+
+	val, err := pipe.Value()
+	require.Nil(t, err)
```

Comment:
suggestion: `require.Nil` checks if given any object is `nil`. But there is `require.NoError` that is specific for error and will produce a bit better output

Reply:
Yes of course :) Thanks

- [ ] require.NoError